modules/nf-core/bedtools/map/main.nf

```diff
@@ -0,0 +1,58 @@
+process BEDTOOLS_MAP {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "bioconda::bedtools=2.30.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/bedtools:2.30.0--hc088bd4_0' :
+        'quay.io/biocontainers/bedtools:2.30.0--hc088bd4_0' }"
+
+    input:
+    tuple val(meta), path(intervals1), path(intervals2)
+    tuple val(meta2), path(chrom_sizes)
+
+    output:
+    tuple val(meta), path("*.${extension}"), emit: map
+    path  "versions.yml"                   , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    extension = intervals1.getExtension()
+    def sizes = chrom_sizes ? "-g ${chrom_sizes}" : ''
+    if ("$intervals1" == "${prefix}.${extension}" ||
+        "$intervals2" == "${prefix}.${extension}")
+        error "Input and output names are the same, use \"task.ext.prefix\" to disambiguate!"
+    """
+    bedtools \\
+        map \\
+        -a $intervals1 \\
+        -b $intervals2 \\
+        $args \\
+        $sizes \\
+        > ${prefix}.${extension}
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        bedtools: \$(bedtools --version | sed -e "s/bedtools v//g")
+    END_VERSIONS
+    """
+
+    stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    extension = tast.ext.suffix ?: "bed"
```

Comment:
Can you change the stub so it works the same as the script part? (the way you get extension here is different)

Reply:
@nvnieuwk done